### PR TITLE
Dev fonc courses

### PR DESCRIPTION
**Pull request by TheoPth**
- Retrieves courses only held by the user
- Use the same data as the token of the um

### CODE REVIEW

File: src/models/login.model.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ const login = {
         if (rows.length > 0) {
           const teacher = P.Teacher.dbToTeacher(rows[0]);
           if (bcrypt.compareSync(password, rows[0].passwordteacher)) {
-            const token = this.generateToken(teacher.id);
+            const token = SToken.generateToken(teacher.firstName, teacher.lastName, 'teacher', 'IG');
```
Comment (yannick-mayeur) [inline src/models/login.model.js]: Why did you remove the id from the token ? the other attributes don't have to be unique so when getting the user based on the token errors might be made.

File: src/models/login.model.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ const login = {
           if (email.endsWith('@etu.umontpellier.fr')) {
             const student = P.Student.dbToStudent(rows[0]);
             if (bcrypt.compareSync(password, rows[0].passwordstudent)) {
-              const token = this.generateToken(student.id);
+              const token = SToken.generateToken(student.firstName, student.lastName, 'student', student.class);
```
Comment (yannick-mayeur) [inline src/models/login.model.js]: same

File: src/middleware/login.middleware.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ */
+const login = async function (req, res, next) {
+  const token = req.headers.authorization;
+  if (token) {
+    const infos = SToken.decryptToken(req.headers.authorization);
+    let user = null;
+    if (infos.role === 'student') {
+      user = await M.Student.getByName(infos.firstname, infos.lastname); 
+    } else if (infos.role === 'teacher') {
+      user = await M.Teacher.getByName(infos.firstname, infos.lastname);
```
Comment (yannick-mayeur) [inline src/middleware/login.middleware.js]: What if firsname, lastname are not unique ?

File: src/models/login.model.js (hunk 1 above)
Comment (TheoPth) [reply src/models/login.model.js]: Because the token from the UM will be exactly with this shape. I changed it to match with the futur connection.

File: src/middleware/login.middleware.js (hunk 3 above)
Comment (TheoPth) [reply src/middleware/login.middleware.js]: It’s unique, thanks to UM connection.

**Comment by yannick-mayeur** [comment]
I have one last comment on the front side of the application, once it is resolved we can merge this :smiley:
Great work!

File: src/middleware/login.middleware.js (hunk 3 above)
Comment (yannick-mayeur) [reply src/middleware/login.middleware.js]: I did not know that. Great !

Comment (yannick-mayeur) [reply src/middleware/login.middleware.js]: Does this mean if we have 2 people called "John Doe", we would have this:

* John Doe
* John Doe2

**Comment by yannick-mayeur** [comment]
Merged
